app/controllers/subscriptions_controller.rb

```diff
@@ -27,31 +27,64 @@ def new
     add_breadcrumb 'New', new_subscription_path
   end
 
+  def show
+    authorize :subscription
+    @subscription = cmr_formatted_subscription_params
+    user = get_subscriber(@subscription['SubscriberId']).fetch(0, {})
+    @user = "#{user['first_name']} #{user['last_name']}"
+    @subscription_concept_id = @subscription['id']
+
+    add_breadcrumb @subscription['id'].to_s
+  end
+
   def create
     authorize :subscription
     # query should be passed as `metadata`
     # subscriber urs_uid should be passed as `user_id`
     # subscriber email should be passed as `email_address`
     @subscription = subscription_params
+    reformated_subscription = reformat_subscription_params_for_submission
+    native_id = "mmt_subscription_#{SecureRandom.uuid}"
 
-    subscription_response = cmr_client.ingest_subscription(reformat_subscription_params_for_submission.to_json, current_user.provider_id, "mmt_subscription_#{SecureRandom.uuid}", token)
+    subscription_response = cmr_client.ingest_subscription(reformated_subscription.to_json, current_user.provider_id, native_id, token)
     if subscription_response.success?
-      redirect_to subscriptions_path, flash: { success: 'Subscription was successfully created.'}
+      flash[:success] = I18n.t('controllers.subscriptions.create.flash.success')
+      redirect_to subscription_path(subscription_response.parsed_body['result']['concept_id'], subscription: reformated_subscription, native_id: native_id)
     else
       Rails.logger.error("Creating a subscription failed with CMR Response: #{subscription_response.clean_inspect}")
-      flash[:error] = subscription_response.error_message
+      flash[:error] = subscription_response.error_message(i18n: I18n.t('controllers.subscriptions.create.flash.success'))
```

Comment:
Should this be controllers.subscriptions.create.flash.error ?

Reply:
Yes